cmd/tealdbg/local_test.go

```diff
@@ -1459,3 +1460,307 @@ func TestGroupTxnIdx(t *testing.T) {
 	a.NoError(err)
 	a.True(pass)
 }
+
+func TestRunAllGloads(t *testing.T) {
+
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+
+	sourceA := `#pragma version 6
+
+	txn ApplicationID
+	bz handle_createapp
+	
+	int 99
+	store 1
+	
+	itxn_begin
+		int acfg
+		itxn_field TypeEnum
+		int 1000000
+		itxn_field ConfigAssetTotal
+		int 3
+		itxn_field ConfigAssetDecimals
+		byte base64 AA== 
+		itxn_field ConfigAssetUnitName
+		byte base64(AAAAAAAAAAA=)
+		itxn_field ConfigAssetName
+		pushbytes 0x0000000000000000 
+		itxn_field ConfigAssetURL
+		global CurrentApplicationAddress
+		dup
+		dup2
+		itxn_field ConfigAssetManager
+		itxn_field ConfigAssetReserve
+		itxn_field ConfigAssetFreeze
+		itxn_field ConfigAssetClawback
+	itxn_submit
+	
+	handle_createapp:
+	int 1`
+
+	sourceB := `#pragma version 6
+
+	txn ApplicationID
+	bz handle_createapp
+	
+	gload 2 1
+	itob
+	log
+	
+	handle_createapp:
+	int 1`
+
+	ops, err := logic.AssembleString(sourceA)
+	a.NoError(err)
+	progA := base64.StdEncoding.EncodeToString(ops.Program)
+
+	ops, err = logic.AssembleString(sourceB)
+	a.NoError(err)
+	progB := base64.StdEncoding.EncodeToString(ops.Program)
+
+	// Transaction group with 5 transactions
+	// 1. Payment txn to app A
+	// 2. Payment txn to app B
+	// 3. App call to app A
+	// 4. App call to app B with gload on app A scratch slot
+	ddrBlob := `{
```

Comment:
@algochoi Another edification question:  What is the significance of exercising `gload` in the PR's context?

* Perhaps mistakenly, I feel like there was a verbal discussion suggesting `gload` + debugger didn't work as intended.  I no longer recall the details.  Given my vague memory, I tried to understand what `gload`-specific changes the PR makes.
* I ran `TestRunAllGloads` against `master` and discovered the test passes once implementing the PR's debugger nil checks.  Which is to say, there's no change specific to `gload`.
* I imagine there's good reason for the test.  But, it's not apparent to me, so I'm asking the question.

Reply:
Currently, `gload` works if you:

1. Supply a dryrun of the entire group transaction
2. Debug every single transaction in the group transaction in sequence, i.e. call `RunAll()` (notably the previous transactions that `gload` refers to needs to be "debugged" so its state and scratch vars are initialized). So putting a `gload` in the `RunAll()` to execute all the transactions in the group seemed interesting to me.

I think `gaid` still has issues because created apps and assets are not initialized properly without using some actual ledger right now.

Reply:
@algochoi Thanks for the context here - makes sense and I assume we'll treat `gaid` out-of-band to the PR.  Resolving.